calico_rkt/calico_rkt.py

```diff
@@ -1,198 +1,212 @@
+        _log.error("This node is not configured for IPv%d. Unassigning IP "
+                   "address %s then exiting." % ip.version, ip)
+        _datastore_client.unassign_address(pool, ip)
         sys.exit(1)
 
     # Create the veth, move into the container namespace, add the IP and
     # set up the default routes.
     ep.mac = ep.provision_veth(Namespace(netns_path), interface)
-    client.set_endpoint(ep)
+    _datastore_client.set_endpoint(ep)
 
-    # Let the caller know what endpoint was created.
     return ep, ip
 
-def _container_remove(hostname, orchestrator_id, container_id, client):
+
+def _container_remove(hostname, orchestrator_id, container_id):
     """
     Remove the indicated container on this host from Calico networking
     """
     # Find the endpoint ID. We need this to find any ACL rules
     try:
-        endpoint = client.get_endpoint(
-            hostname=hostname,
-            orchestrator_id=orchestrator_id,
-            workload_id=container_id
-        )
+        endpoint = _datastore_client.get_endpoint(hostname=hostname,
+                                                  orchestrator_id=orchestrator_id,
+                                                  workload_id=container_id)
     except KeyError:
-        print_stderr("Container %s doesn't contain any endpoints" % container_id)
+        _log.error(
+            "Container %s doesn't contain any endpoints" % container_id)
         sys.exit(1)
 
     # Remove any IP address assignments that this endpoint has
     for net in endpoint.ipv4_nets | endpoint.ipv6_nets:
         assert(net.size == 1)
-        client.unassign_address(None, net.ip)
+        _datastore_client.unassign_address(None, net.ip)
 
     # Remove the endpoint
     netns.remove_veth(endpoint.name)
 
     # Remove the container from the datastore.
-    client.remove_workload(hostname, orchestrator_id, container_id)
+    _datastore_client.remove_workload(hostname=hostname,
+                                      orchestrator_id=orchestrator_id,
+                                      workload_id=container_id)
 
-    print_stderr("Removed Calico interface from %s" % container_id)
+    _log.info("Removed Calico interface from %s" % container_id)
 
-def _create_profile(endpoint, profile_name, ip, client):
-    """
-    Configure the calico profile for a pod.
 
-    Currently assumes one pod with each name.
+def _set_profile_on_endpoint(endpoint, profile_name, ip):
     """
-    print_stderr('Configuring Pod Profile: %s' % profile_name)
+    Configure the calico profile to the endpoint
+    """
+    _log.info('Configuring Pod Profile: %s' % profile_name)
 
-    if client.profile_exists(profile_name):
-        print_stderr("Profile with name %s already exists, applying to endpoint." % (profile_name))
+    if _datastore_client.profile_exists(profile_name):
+        _log.info(
+            "Profile with name %s already exists, applying to endpoint." % (profile_name))
```

Comment:
This branch of the if statement doesn't seem extremely important, so lets invert the statement and just leave the True branch out
```
if not _datastore_client.profile_exists(profile_name):
    _log.info("Creating profile %s." % (profile_name))
    _datastore_client.create_profile(profile_name)
```